File: android/jsbridge/src/main/java/net/kibotu/jsbridge/commands/BridgeAware.kt
package net.kibotu.jsbridge.commands

import net.kibotu.jsbridge.JavaScriptBridge

/**
 * Opt-in interface for commands that need access to the [JavaScriptBridge] instance.
 *
 * The bridge is set automatically by [JavaScriptBridge.inject] after construction,
 * so commands can use no-arg constructors and still access the bridge (or its context)
 * at execution time.
 */
interface BridgeAware {
    var bridge: JavaScriptBridge?
}

File: ios/JSBridge/Sources/JSBridge/BridgeAware.swift
import Foundation

/// Opt-in protocol for commands that need access to the bridge instance.
///
/// The bridge is set automatically by ``JavaScriptBridge/init(webView:viewController:bridgeName:commands:)``
/// after construction, so commands can use no-arg initializers and still access the bridge
/// (or its `webView` / `viewController`) at execution time.
///
/// `@unchecked Sendable` is expected on adopters because the mutable `bridge` property
/// is only accessed on `@MainActor`.
@MainActor
public protocol BridgeAware: AnyObject {
    var bridge: JavaScriptBridge? { get set }
}
